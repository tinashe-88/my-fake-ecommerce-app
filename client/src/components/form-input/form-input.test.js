import React from 'react'
import { shallow } from 'enzyme'

import { FormInput } from './form-input'

describe('FormInput component', () => {
  let wrapper
  let mockHandleChange

  beforeEach(() => {
    mockHandleChange = jest.fn()

    const mockProps = {
      label: 'email',
      value: '123@email.com',
      handleChange: mockHandleChange
    }
    
    wrapper = shallow(<FormInput {...mockProps}/>)
  })

  it('Should render FormInput component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should call handleChange method when the input changes', () => {
    wrapper.find('FormInputContainer').simulate('change')
    expect(mockHandleChange).toHaveBeenCalled()
  })

  it('Should render FormInputLabel if a label exists', () => {
    expect(wrapper.exists('FormInputLabel')).toBe(true)
  })

  it('Should NOT render FormInputLabel if no label exists', () => {
    const mockNewProps = {
      label: '',
      value: '123@email.com',
      handleChange: mockHandleChange
    }

    const newWrapper = shallow(<FormInput {...mockNewProps}/>)

    expect(newWrapper.exists('FormInputLabel')).toBe(false)
  })
})