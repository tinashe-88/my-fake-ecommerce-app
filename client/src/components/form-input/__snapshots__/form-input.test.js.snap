// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component Should render FormInput component 1`] = `
<styled.div>
  <FormInputContainer
    onChange={[MockFunction]}
    value="123@email.com"
  />
  <FormInputLabel
    className="shrink form-input-label"
  >
    email
  </FormInputLabel>
</styled.div>
`;
